src/mjolnir/pbfgraphparser.cc

```diff
@@ -616,15 +616,20 @@ struct graph_callback : public OSMPBF::Callback {
       } */
       else if (tag.first == "official_name" && !tag.second.empty()) {
         w.set_official_name_index(osmdata_.name_offset_map.index(tag.second));
-
       } else if (tag.first == "max_speed") {
         try {
-          max_speed = std::stof(tag.second);
+          if (tag.second == "unlimited") {
+            // this way has an unlimited speed limit (german autobahn)
+            max_speed = kUnlimitedSpeedLimit;
```

Comment:
I think you do want to mark this as a tagged speed; otherwise, it will try to enhance it.  see https://github.com/valhalla/valhalla/blob/master/src/mjolnir/graphenhancer.cc#L120-L131

Reply:
👍 thanks for catching that. Updated.